doc/concepts/morphology.rst

```diff
@@ -3,56 +3,117 @@
-joined together to form a tree. Only :ref:`cable cells <modelcablecell>` support custom
-morphologies in Arbor.
+In morphologically detailed simulations of neurons,
+`the cell is modelled <https://en.wikipedia.org/wiki/Cable_theory>`_
+as a set of round electrical cables, which have, among others properties,
+capacitances and resistances. These can in turn be expressed as a system of
+partial differential equations which can be solved to compute a current or
+voltage. The geometry of the biological cell is approximated as a tree of
+cable segments, which is a :term:`morphology` in Arbor nomenclature.
 
-The building blocks of a morphology tree are points and segments.
-A *point* is a three-dimensional location and a radius, used to mark the cross-sectional
-centre and radius of the cable.
+Let's first define the building blocks out of which we can construct a
+proper definition for a morphology:
 
-.. csv-table::
+.. glossary::
+
+  mpoint
+    A point in 3D space has three coordinates. In Arbor, we add a fourth coordinate: radius.
+
+.. csv-table:: The properties of a :term:`point <mpoint>`, in the context of cable cell morphologies.
    :widths: 10, 10, 30
+   :align: center
 
    **Field**,   **Type**, **Description**
-   ``x``,       real, x coordinate of centre of cable (μm).
-   ``y``,       real, y coordinate of centre of cable (μm).
-   ``z``,       real, z coordinate of centre of cable (μm).
+   ``x``,       real, x coordinate of center of cable (μm).
+   ``y``,       real, y coordinate of center of cable (μm).
+   ``z``,       real, z coordinate of center of cable (μm).
    ``radius``,  real, cross sectional radius of cable (μm).
 
+.. glossary::
 
-A *segment* is a frustum (cylinder or truncated cone), with the centre and radius at each
-end defined by a pair of points. In other words, in Arbor the radius between two points is interpolated
-linearly, resulting in either a cylinder (equal radii) or truncated cone (differing radii),
-centred at the line through the pair of points.
+  segment
+    A segment is a frustum (cylinder or truncated cone), with the center and radius at each
+    end defined by a pair of :term:`points <mpoint>`. In other words, in Arbor the radius between two points is interpolated
+    linearly, resulting in either a cylinder (equal radii) or truncated cone (differing radii),
+    centred at the line through the pair of points.
 
-.. csv-table::
+.. csv-table:: The properties of a :term:`segment`.
    :widths: 10, 10, 30
+   :align: center
 
-   **Field**,   **Type**, **Description**
+   **Field**,      **Type**,                           **Description**
    ``prox``,       :py:class:`point <arbor.mpoint>`,   the center and radius of the proximal end.
    ``dist``,       :py:class:`point <arbor.mpoint>`,   the center and radius of the distal end.
-   ``tag``,        integer, tag meta-data.
+   ``tag``,        integer (:term:`tag`),              "differentiate soma, axon, dendrite, etc."
```

Comment:
I think the description of the tag is quite limiting. Maybe we can say: 
"tag meta-data, can be used to classify segments of the same kind (ex: soma, dendrite, but also arbitrary use-defined groups)"

Reply:
:+1: